### ISSUE TITLE
Mediaserver y canales que importan xbmc*

### ISSUE THREAD

**Issue by Palatosino** [tvalacarta/pelisalacarta#376]
Hay canales que hacen import xbmc o xbmcgui, lo que hace que no puedan ser usados en la versión mediaserver:

grep xbmcgui * | cut -d: -f1 | sort -u
aquitorrent.py
bityouth.py
bricocine.py
cinefox.py
configuracion.py
crimenes.py
multicineonline.py
peliculasdk.py
pordede.py
txibitsoft.py
verseriesonlinetv.py
zentorrents.py

**Comment by SeiTaN** [comment]
En configuración se puede arreglar cambiando.
```python
import xbmcgui
            yes_pressed = xbmcgui.Dialog().yesno( "Versión "+version+" disponible" , "¿Quieres instalarla?" )
```
Por....
```python
from platformcode import platformtools
            yes_pressed = platformtools.dialog_yesno( "Versión "+version+" disponible" , "¿Quieres instalarla?" 
```

**Comment by tvalacarta** [comment]
Cierto :(

De todas formas eso no impide que el canal funcione, falla esa opción pero nada más.

He visto que por ejemplo en pordede es el mismo caso, funcionará bien en la versión mediaserver aunque tenga el literal "xbmcgui"

**Comment by SeiTaN** [comment]
No he comprobado el resto de casos y no se que funcionalidad tendrá esas llamadas a xbmc. si hacen cosas que no estén disponibles en platformtools (modulo común), se podría poner un try except.
```python

try:
    import xbmc
except ImportError:
  xbmc = None


if xbmc:
   xbmc.funcion
   asdadsfasdf
   asfdasdf
   ...
```

**Comment by CmosGit** [comment]
Bueno, que importen xbmc o xbmcgui no significa que den error en mediaserver. Por ejemplo cinefox utiliza ambos, uno en un método para refrescar el itemlist y otro para elegir un número de página a cargar, y ambas opciones están controladas y solo se intenta usarlas si la plataforma es Kodi/xbmc. Habría que mirar cada caso...

**Comment by Palatosino** [comment]
Le estoy dando un vistazo más concreto.

Veo que el problema concreto son canales que hacen:
class TextBox1( xbmcgui.WindowDialog ):

Todos ellos fallan. Son:
aquitorrent.py
bityouth.py
bricocine.py
multicineonline.py
peliculasdk.py
pordede.py
txibitsoft.py
verseriesonlinetv.py
zentorrents.py

**Comment by superberny70** [comment]
Eso si es una dependencia clara entre el canal y la plataforma.
Jesus corrigeme si me equivoco, pero originalmente la funcion isGeneric() servia para identificar los canale genericos (sin importaciones de las librerias xbmc) de los q eran exclusivos de xbmc.
Lo correcto seria no utilizar las librerias xbmc, por eso creamos plataformtools. Para emular en cada plataforma un sustituto a las funciones de esas librerias.
Cuando no sea posible usar plataformtools lo q habria q hacer como dice @CmosGit es controlar el error q se producira al importarlas.

Otra dependencia es el uso de etiquetas de formato tipo [COLOR RED] en los titulos o sinopsis, por ejemplo.

Respecto a la clase class TextBox1( xbmcgui.WindowDialog ) sus creadores deberian decidir que hacer con ella. Yo ni la he usado, ni la usare nunca.

**Comment by tvalacarta** [comment]
De este tema (perdón por la demora en opinar) hablamos @neno1978 y yo hace tiempo.

Lo que habría que hacer es declarar de alguna manera que un canal sólo va a funcionar en cierta plataforma, como es el caso de sus canales que sólo funcionan en Kodi y esa es su decisión.

No llegamos a ver cómo identificar esto en el xml del canal, pero tampoco sería difícil.

Lógicamente no está bien incluir código propio de Kodi en un canal ya que no funcionará en otras plataformas, pero tampoco podemos pedir a alguien que aporta un canal que lo pruebe en una plataforma que ni conoce ni le interesa.

Si alguien se anima a corregir el canal para evitar el error del import está bien, pero igualmente esos canales probablemente no funcionarán fuera de Kodi por una u otra razón.

**Comment by superberny70** [comment]
Creo q no cuesta nada hacerlos genericos. Solo hay q evitar una serie de cosas (para esto estaria bien una pequeña guia de estilo: "evitar importar de xbmc"; "nada de '{0}'.format(...) "; etc..)

No nos engañemos la mayoria utilizamos Kodi y programamos nuestros canales sobre kodi, si se permiten canales exclusivos para kodi alguien podria pensar: _"¿para q me voy a molestar en hacerlo multiplataforma y exponerme a q se me quejen los usuarios de plex por ejemplo? A mis canales le pongo 'SOLO KODI' y me quito problemas"_

Es una opinion pero creo q si quieres mantener la compatibilidad no deberias ir por ese camino.

**Comment by SeiTaN** [comment]
Estoy de acuerdo con las dos opiniones, pero creo que al final depende del autor, por el tema que ha dicho @Palatosino, `class TextBox1( xbmcgui.WindowDialog ):` ¿esto se puede migrar a otra plataforma? ¿que hace exactamente?

**Comment by tvalacarta** [comment]
Yo creo que no son excluyentes.

Hay que intentar que todos los canales sean multiplataforma, con guías, revisiones, etc. Es lo mejor sin duda.

Pero también es bueno que alguien con poca experiencia pueda aportar un canal, aunque no tenga ni idea de lo que es Plex o no le interese.
